Tools/autotest/arduplane.py

```diff
@@ -1755,6 +1755,96 @@ def fly_terrain_mission(self):
 
         self.fly_mission("ap-terrain.txt", mission_timeout=600)
 
+    def ekf_lane_switch(self):
+
+        # new lane swtich available only with EK3
+        self.set_parameter("EK3_ENABLE", 1)
+        self.set_parameter("EK2_ENABLE", 0)
+        self.set_parameter("AHRS_EKF_TYPE", 3)
+        self.set_parameter("EK3_AFFINITY", 15)
+        self.set_parameter("GPS_TYPE2", 1)
+        self.set_parameter("SIM_GPS2_DISABLE", 0)
+        self.set_parameter("SIM_BARO2_DISABL", 0)
+        self.set_parameter("SIM_BARO_COUNT", 2)
+
+        # some parameters need reboot to take effect
+        self.reboot_sitl()
+        
+        # get flying
+        self.load_mission('CMAC-soar.txt')
+        self.change_mode('AUTO')
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+        self.delay_sim_time(10)
+
+        self.context_push()
+        ex = None
+        try:
+            # start IMU tests
+            self.progress("Checking Lane Switch with Accelerometer")
+            # sudden acceleration in x - axis
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("INS_ACCOFFS_X", 5))
+            self.set_parameter("INS_ACCOFFS_X", 0)
+            self.delay_sim_time(10)
```

Comment:
Why the flat delays here?  If it's to "let things settle", it would be nice to detect that settling if possible rather than assume it happens within 10 seconds.  Do we provide enough information via mavlink to detect?

Reply:
i changed it to wait for plane to complete one circle